### PR TITLE
DOCS 935: Clarify type usage between app and modular resources docs

### PR DESCRIPTION
**Pull request by sguequierre**
* Add/update glossary definitions for type, subtype, and model
* Update glossary tooltips

### CODE REVIEW

File: docs/appendix/glossary/model.md [hunk 1]
```diff
@@ -2,9 +2,10 @@
...
-short_description: A particular implementation of a component type. For example, UR5e is a model of the arm component type.
+short_description: A particular implementation of a resource. For example, UR5e is a model of the arm component resource subtype's API.
 aka:
 ---
 
-A particular implementation of a {{< glossary_tooltip term_id="component" text="component" >}} type.
-For example, UR5e is a model of the arm component type.
+A particular implementation of a {{< glossary_tooltip term_id="resource" text="resource" >}} {{< glossary_tooltip term_id="subtype" text="subtype" >}}.
+
+For example: UR5e is a *model* of arm, implementing the [resource API](/program/apis/) of the [arm](/components/arm/) subtype of {{< glossary_tooltip term_id="component" text="component" >}}, which, in turn, is a {{< glossary_tooltip term_id="type" text="type" >}} of resource.
```
Comment (sguequierre) [inline docs/appendix/glossary/model.md:11]:
```suggestion
For example: `ur5e` is a *model* of arm, implementing the [resource API](/program/apis/) of the [arm](/components/arm/) subtype of {{< glossary_tooltip term_id="component" text="component" >}}, which, in turn, is a {{< glossary_tooltip term_id="type" text="type" >}} of resource.
```

File: docs/appendix/glossary/type.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+title: Type
```
Comment (sguequierre) [inline docs/appendix/glossary/type.md:2]:
```suggestion
title: Type (Resource API)
```
This longer title is only really necessary if we end up wanting to make the configuration/more normal usage of type a separate glossary, like in your pr, & separate that out more! @JessamyT

Comment (JessamyT) [reply docs/appendix/glossary/type.md:2]: Yeah I'm cool with this just being "type" based on the way you've done this PR!

File: docs/appendix/glossary/subtype.md
Comment (JessamyT) [inline docs/appendix/glossary/subtype.md:1]: I'm getting a little hung up on "component resource type." I'm wondering if it could be interpreted in too many different ways. What do you think of "...an arm is a subtype of the component type" without "resource" in there? I'm thinking it might be easier to parse.

**Review by JessamyT — APPROVED** [review]
This looks really good to me based on the current uses of "type" and "subtype." Nice job.

I still think this is a confusing state of affairs so I'm adding it to discuss tomorrow, but regardless I think this is a great clarification for how things are right now!

File: docs/appendix/glossary/subtype.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+---
+title: Subtype
+id: subtype
+full_link:
+short_description: A particular implementation of a resource type. For example, an arm is a subtype of the component resource type.
+aka:
+---
+
+A particular implementation of a {{< glossary_tooltip term_id="type" text="type" >}} of {{< glossary_tooltip term_id="resource" text="resource" >}}, as delineated by its {{< glossary_tooltip term_id="api-namespace-triplet" text="api-namespace-triplet" >}}
```
Comment (JessamyT) [inline docs/appendix/glossary/subtype.md:9]:
```suggestion
A category within a {{< glossary_tooltip term_id="type" text="type" >}} of {{< glossary_tooltip term_id="resource" text="resource" >}}.
All resources of a given subtype share the same API.
Models are subcategories within that subtype, implementing the common API.
```
"as delineated by its api-namespace-triplet" makes this first sentence hard to parse. Let's introduce them to the concept in a more concrete way before getting into that more technical piece (see suggestion below for where to put it).

File: docs/appendix/glossary/subtype.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+For example, an arm is a subtype of the {{< glossary_tooltip term_id="component" text="component" >}} resource type, while the `ur5e` is a {{< glossary_tooltip term_id="model" text="model" >}} of the arm subtype's API.
+
+The [Vision Service](/services/vision/) is a subtype of the {{< glossary_tooltip term_id="service" text="service" >}} resource type.
```
Comment (JessamyT) [inline docs/appendix/glossary/subtype.md:13]:
```suggestion
The [Vision Service](/services/vision/) is a subtype of the {{< glossary_tooltip term_id="service" text="service" >}} resource type.

A subtype is designated by its {{< glossary_tooltip term_id="api-namespace-triplet" text="api-namespace-triplet" >}}.
```

**Review by JessamyT — CHANGES_REQUESTED** [review]
circled back to "subtype" and added some comments

File: docs/appendix/glossary/subtype.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+---
+title: Subtype
+id: subtype
+full_link:
+short_description: A particular implementation of a resource type. For example, an arm is a subtype of the component resource type.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/appendix/glossary/subtype.md:5]: I like the idea here but I don't think the second sentence is easy to parse, maybe 
```suggestion
short_description: A group of component or service models that share the same type of resource API. For example, an arm is a subtype of the component resource type.
```

File: docs/appendix/glossary/subtype.md (hunk 3 above)
Comment (sguequierre) [reply docs/appendix/glossary/subtype.md:9]: I agree less technical but to keep some of that phrasing in mind for people maybe 
```suggestion
A category within a {{< glossary_tooltip term_id="type" text="type" >}} of {{< glossary_tooltip term_id="resource" text="resource" >}}.
Resources belonging to a subtype share the same API.
{{< glossary_tooltip term_id="model" text="Models" >}} implement that subtype's API protocol.
``` 
?

File: docs/appendix/glossary/subtype.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JessamyT) [reply docs/appendix/glossary/subtype.md:5]: Oops I left the word "the" in before "component." Just edited to take it out.

File: docs/appendix/glossary/subtype.md (hunk 3 above)
Comment (JessamyT) [reply docs/appendix/glossary/subtype.md:9]: Okay sure! 
```suggestion

A category within a {{< glossary_tooltip term_id="type" text="type" >}} of {{< glossary_tooltip term_id="resource" text="resource" >}}.
Resource models belonging to a subtype share the same API.
{{< glossary_tooltip term_id="model" text="Models" >}} implement that subtype's API protocol with different drivers.
```
I think adding "models" to the second line will help too.

**Review by JessamyT — APPROVED** [review]

**Review by npentrel — CHANGES_REQUESTED** [review]
Let's hold off on merging this until we conclude our docs-wide discussion on it

File: docs/extend/create/create-subtype.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    - [<file>buf.lock</file>](https://buf.build/docs/configuration/v1/buf-lock/)
+
+6. In the <file>/src/</file> directory of your module, use the protobuf compiler to [generate](https://buf.build/docs/tutorials/getting-started-with-buf-cli/#generate-code) all other necessary protocol buffer code, based on the `<subtype name>.proto` file you wrote.
+
+    - [Example generated files for a Python-based service](https://github.com/viam-labs/speech/tree/main/src/proto)
+      - (<file>speech.proto</file> was manually written, not generated.)
+
+## Next steps
+
+Now that your resource API is defined, [create a new model](/extend/modular-resources/create/#code-a-new-resource-model) that implements your new API.
```
Comment (sguequierre) [inline docs/extend/create/create-subtype.md:69]:
```suggestion
Now that your resource API is defined, [create a new model](/extend/modular-resources/create/#code-a-new-resource-model) that implements your new API.

```

**Comment by JessamyT** [comment]
It looks like some strange merge stuff happened here--bonus stuff is now part of the diff
